sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/Listeners/ServiceBusListenerTests.cs

```diff
@@ -67,23 +67,21 @@ public ServiceBusListenerTests()
         [Test]
         public async Task ProcessMessageAsync_Success()
         {
-            var message = new CustomMessage();
-            var systemProperties = new Message.SystemPropertiesCollection();
-            typeof(Message.SystemPropertiesCollection).GetProperty("SequenceNumber").SetValue(systemProperties, 1);
-            typeof(Message.SystemPropertiesCollection).GetProperty("DeliveryCount").SetValue(systemProperties, 55);
-            typeof(Message.SystemPropertiesCollection).GetProperty("EnqueuedTimeUtc").SetValue(systemProperties, DateTime.Now);
-            typeof(Message.SystemPropertiesCollection).GetProperty("LockedUntilUtc").SetValue(systemProperties, DateTime.Now);
-            typeof(Message).GetProperty("SystemProperties").SetValue(message, systemProperties);
-
-            message.MessageId = Guid.NewGuid().ToString();
-            _mockMessageProcessor.Setup(p => p.BeginProcessingMessageAsync(message, It.IsAny<CancellationToken>())).ReturnsAsync(true);
+            var message = ServiceBusModelFactory.ServiceBusReceivedMessage(messageId: Guid.NewGuid().ToString());
+            typeof(ServiceBusReceivedMessage).GetProperty("SequenceNumber").SetValue(message, 1);
```

Comment:
Can't we do this without reflection?

Reply:
Technically, yes, but these are just tests.

Reply:
It's a sign you need to add more ModelFactory methods to the service bus.

Reply:
nevermind this is already covered by our existing factory method. Will update.